### PR TITLE
`Development`: Switch postgres e2e setup to Jenkins/Gitlab

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

We are currently only testing the E2E test with the Atlassian suite (bamboo, bitbucket, jira). 

### Description

This PR switches the postgres E2E plan to using Gitlab and Jenkins. It also fixes the gitlab health check and adds local user management (necessary users are created before each run). 

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
This PR makes the server tests running for over one hour and probally get stuck somewhere.

File: src/test/cypress/support/users.ts [hunk 1]
```diff
@@ -1,5 +1,27 @@
 import { BASE_API, GET, USER_ID_SELECTOR } from './constants';
 
+export enum UserRole {
+    Instructor = 'ROLE_INSTRUCTOR',
+    Tutor = 'ROLE_TA',
+    Student = 'ROLE_USER',
+}
```
Comment (pal03377) [inline src/test/cypress/support/users.ts:7]:
```suggestion
};
```

**Review by pal03377 — DISMISSED** [review]
Code looks good! Also, the health status of GitLab will be fixed 🎉

**Review by pal03377 — DISMISSED** [review]

**Review by pal03377 — APPROVED** [review]

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Pull request merged**